rocm-related: misc changes

Madouura opened the pull request:
###### Description of changes
Tracking: #197885
Supercedes: #203335
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Madouura committed:
rocm-related: build documentation by default

miopen: build documentation by default

miopengemm: build documentation by default

rocwmma: build documentation by default

Madouura committed:
miopen: don't use nativeBuildInputs in test fixup

Madouura committed:
rocm-related: use empty list instead of null for gpuTargets

composable_kernel: use empty list instead of null for gpuTargets

rocwmma: use empty list instead of null for gpuTargets

Madouura committed:
rocm-related: remove ? null and related asserts

composable_kernel: remove ? null and related asserts

hipcub: remove ? null and related asserts

hipsparse: remove ? null and related asserts

miopen: remove ? null and related asserts

miopengemm: remove ? null and related asserts

rccl: remove ? null and related asserts

rocblas: remove ? null and related asserts

rocfft: remove ? null and related asserts

rocprim: remove ? null and related asserts

rocrand: remove ? null and related asserts

rocsparse: remove ? null and related asserts

rocthrust: remove ? null and related asserts

rocwmma: remove ? null and related asserts

Madouura committed:
rocm-related: optional GITHUB_TOKEN for update script

hip: optional GITHUB_TOKEN for update script

rocm-llvm: optional GITHUB_TOKEN for update script

clang-ocl: optional GITHUB_TOKEN for update script

hipcub: optional GITHUB_TOKEN for update script

hipsparse: optional GITHUB_TOKEN for update script

miopen: optional GITHUB_TOKEN for update script

miopengemm: optional GITHUB_TOKEN for update script

rccl: optional GITHUB_TOKEN for update script

rocblas: optional GITHUB_TOKEN for update script

rocclr: optional GITHUB_TOKEN for update script

rocfft: optional GITHUB_TOKEN for update script

rocm-comgr: optional GITHUB_TOKEN for update script

rocm-device-libs: optional GITHUB_TOKEN for update script

rocm-opencl-runtime: optional GITHUB_TOKEN for update script

rocm-runtime: optional GITHUB_TOKEN for update script

rocm-thunk: optional GITHUB_TOKEN for update script

rocmlir: optional GITHUB_TOKEN for update script

rocprim: optional GITHUB_TOKEN for update script

rocrand: optional GITHUB_TOKEN for update script

rocsparse: optional GITHUB_TOKEN for update script

rocthrust: optional GITHUB_TOKEN for update script

rocwmma: optional GITHUB_TOKEN for update script

tensile: optional GITHUB_TOKEN for update script

rocm-cmake: optional GITHUB_TOKEN for update script

rocminfo: optional GITHUB_TOKEN for update script

rocm-smi: optional GITHUB_TOKEN for update script

Madouura commented:
Please do review, but do not merge until I figure out what's going on with ``nixpkgs-review``.
This is a new error now only showing and I suspect it might have to do with what SuperSandro2000 commented on about ``strictDeps``.

Madouura committed:
rocm-related: lib.strings -> lib

composable_kernel: lib.strings -> lib

rocblas: lib.strings -> lib

rocwmma: lib.strings -> lib

SuperSandro2000 commented on pkgs/development/libraries/rocwmma/default.nix:
```diff
@@ -6,31 +6,24 @@
-
 # Building tests isn't working for now
 # undefined reference to symbol '_ZTIN7testing4TestE'
 assert buildTests == false;
```
I think this should be moved to meta.broken

Madouura replied:
Because the tests don't work?
You mean something like ``broken = (!buildTests)``?

SuperSandro2000 replied:
yes

Madouura commented:
Result of `nixpkgs-review pr 203412` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>25 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocrand</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>rocwmma</li>
  </ul>
</details>

Madouura committed:
rocm-related: move asserts to broken

composable_kernel: move asserts to broken

rocblas: move asserts to broken

rocthrust: move asserts to broken

rocwmma: move asserts to broken

Madouura commented:
One moment while I deprecate ``rocmVersion`` and ``repoVersion``...

Madouura committed:
rocm-related: deprecate rocmVersion and repoVersion

clang-ocl: deprecate rocmVersion and repoVersion

hipcub: deprecate rocmVersion and repoVersion

hipsparse: deprecate rocmVersion and repoVersion

miopen: deprecate rocmVersion and repoVersion

miopengemm: deprecate rocmVersion and repoVersion

rccl: deprecate rocmVersion and repoVersion

rocblas: deprecate rocmVersion and repoVersion

rocfft: deprecate rocmVersion and repoVersion

rocmlir: deprecate rocmVersion and repoVersion

rocprim: deprecate rocmVersion and repoVersion

rocrand: deprecate rocmVersion and repoVersion

rocsparse: deprecate rocmVersion and repoVersion

rocthrust: deprecate rocmVersion and repoVersion

rocwmma: deprecate rocmVersion and repoVersion

tensile: deprecate rocmVersion and repoVersion

Madouura commented:
It'll be a bit to compile everything again, but unless there are other issues, this should be good for review and merge.

Madouura commented:
Result of `nixpkgs-review pr 203412` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>26 packages built:</summary>
  <ul>
    <li>blender-hip</li>
    <li>clang-ocl</li>
    <li>composable_kernel</li>
    <li>hip</li>
    <li>hipcub</li>
    <li>hipsparse</li>
    <li>llvmPackages_rocm.clang</li>
    <li>llvmPackages_rocm.clangNoCompilerRt</li>
    <li>llvmPackages_rocm.llvm</li>
    <li>miopengemm</li>
    <li>rccl</li>
    <li>rocblas</li>
    <li>rocclr</li>
    <li>rocm-comgr</li>
    <li>rocm-device-libs</li>
    <li>rocm-opencl-icd</li>
    <li>rocm-opencl-runtime</li>
    <li>rocm-runtime</li>
    <li>rocminfo</li>
    <li>rocmlir</li>
    <li>rocprim</li>
    <li>rocrand</li>
    <li>rocsparse</li>
    <li>rocthrust</li>
    <li>rocwmma</li>
    <li>tensile</li>
  </ul>
</details>

Flakebi reviewed (commented):
Looks good from my side
